docs/guide/release.md

```diff
@@ -0,0 +1,132 @@
+# Releasing a new version
+
+Releasing a new version of the SIX Web Components is a task performed by members of the Core Team,
+and upon discussion with other team members. The following document outlines the different steps
+required.
+
+It is recommended that you read through this document in its entirety before you start with the
+process!
+
+## Preparation
+
+Before you start with the release, there are a few steps that need to be undertaken.
+
+### Merging PRs
+
+Before you start, you need to make sure that all PRs that need to go in this release are merged!
+Check the list of open PRs [here](https://github.com/six-group/six-webcomponents/pulls), and chase
+down the authors if needed.
+
+### Test changes locally
+
+To avoid issues once the process has been started, it is good practice to test the changes locally:
+
+1. Checkout `main`
+2. Navigate to the root of the project
+3. Run `npm ci`. This will ensure that the required packages are correctly installed
+4. Run `npm run build`. This step will build all the different subprojects and make sure there are
+   no errors.
+
+If the steps above complete with no error(s), the release can be considered ready.
+
+### Choose version number
+
+The next step consists in choosing the version number under which this new release will be made.
+Although it should be discussed beforehand with other team members, we follow semantic versioning.
+
+Choose a version number that makes sense for the current changelog. We will call it `VERSION_NUMBER`
+for the rest of this document. Whenever you see a `VERSION_NUMBER` placeholder, you should
+substitute it with your version
+
+### Create preparation branch
+
+Before the release is ready to be created, the changelog need to be updated. Start by creating a
+branch from `main` using the following name:
+
+```
+feature/prepare-M-m-B-release
+```
+
+where:
+
+- `M`: Major version identifier
+- `m`: Minor version identifier
+- `B`: Bugfix version identifier
+
+As an example: if in the previous step you decided the new version is `4.2.3`, then the branch name
+is:
+
+```
+feature/prepare-4-2-3-release
+```
+
+You now need to check that the [`changelog.md`](../changelog.md) file is correctly formatted and
+that all changes listed are in fact merged.
+
+Edit the `changelog.md` and create an entry with the changes. The title should follow the format
+
+```
+## VERSION_NUMBER - YYYY-MM-dd
+```
+
+So, for the above example, and assuming the date is January 10th, 2024:
+
+```
+## 4.2.3 - 2024-01-10
+```
+
+Also, do not forget to add a blank `Upcoming` section for future changes.
+
+Push, and open a PR to merge with `main` with the following title/description:
+
+- Title: `docs: Update changelog for VERSION_NUMBER release`
+- Description:
+
+```
+### ❓ Type of change
```

Comment:
Great description of the release docs, I like it thanks!
However I wouldn't put it in the docs itself as the users would not need to know it. I suggest to keep it as a readme or markdown file in a separate section outside of the docs. 

Suggestion: Alongside the contributing.md file

what do you think @ibirrer

Reply:
Yes, we could simply move it to the root folder